### PR TITLE
"image/jpg" -> "image/jpeg" on MIME type checks and saving attachments

### LINKED ISSUE #4602: Saved attachments with contentType "image/jpg" named with ".attach" extension
In `SaveAttachmentTask.java`, the call to `mimeTypeMap.getExtensionFromMimeType(contentType)` with the type `image/jpg` returns `null` on devices with the AOSP version of `MimeUtils.java`.  This is because the Android platform doesn't recognize `image/jpg` as a valid MIME type.

With the `image/jpeg` type, however, an extension is returned as expected.

[`MimeUtils.java`](https://android.googlesource.com/platform/libcore/+/android-6.0.0_r5/luni/src/main/java/libcore/net/MimeUtils.java) (AOSP)

        add("image/ief", "ief");
        // add ".jpg" first so it will be the default for guessExtensionFromMimeType
        add("image/jpeg", "jpg");
        add("image/jpeg", "jpeg");
        add("image/jpeg", "jpe");
        add("image/pcx", "pcx");

I've dealt with issues in the past where device manufacturers have edited this map (love you, Samsung ㅗ\*.*ㅗ), which may result in sending devices using content types which receiving devices cannot look up via `getExtensionFromMimeType`.  Where to put a sanitizer might be a bit tricky.  I'll dig deeper tomorrow.

### PR DESCRIPTION
**Pull request by unrulygnu**
Fixes #4602 

Also facilitates quick fixes for similar issues resulting from OEM changes to [`MimeUtils.java`](https://android.googlesource.com/platform/libcore/+/android-6.0.0_r5/luni/src/main/java/libcore/net/MimeUtils.java).

### CODE REVIEW

**Comment by McLoo** [comment]
Did test yet. Does forwarding still work?

**Comment by unrulygnu** [comment]
Forwarding still works, but double-checking in the debugger showed me a couple of more spots to apply the fix.  Thanks!

File: src/org/thoughtcrime/securesms/util/MediaUtil.java [hunk 1]
```diff
@@ -76,7 +76,20 @@ public static Slide getSlideForAttachment(Context context, Attachment attachment
...
-    return type;
+    return getCorrectedMimeType(type);
+  }
+
+  public static @Nullable String getCorrectedMimeType(@Nullable String mimeType) {
+    if (mimeType == null) return null;
+
+    switch(mimeType) {
+    case "image/jpg":
+      return MimeTypeMap.getSingleton().hasMimeType(ContentType.IMAGE_JPEG)
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/util/MediaUtil.java]: is this ever false?

Comment (unrulygnu) [inline src/org/thoughtcrime/securesms/util/MediaUtil.java]: OEM's mangle  [`MimeUtils.java`](https://android.googlesource.com/platform/libcore/+/android-6.0.0_r5/luni/src/main/java/libcore/net/MimeUtils.java) in mysterious ways.  If they omit `image/jpeg` in favor of `image/jpg`, our forcing `image/jpeg` will cause issues on the local device (`.attach`, etc.).

Comment (mcginty) [inline src/org/thoughtcrime/securesms/util/MediaUtil.java]: Oh, Android... what a mess you are. Ok cool better safe than sorry.

**Comment by mcginty** [comment]
lgtm

File: src/org/thoughtcrime/securesms/util/MediaUtil.java [hunk 2]
```diff
@@ -76,7 +76,20 @@ public static Slide getSlideForAttachment(Context context, Attachment attachment
...
+             ? ContentType.IMAGE_JPEG
+             : mimeType;
+    default:
```
Comment (gryder0803) [inline src/org/thoughtcrime/securesms/util/MediaUtil.java:90]: Sorrow and love ❤️ I know you will love it but you have a lot to say

**Pull request closed without merging**